tests/unit/utilities/will-interrupt-process-test.js

```diff
@@ -160,16 +160,43 @@ describe('will interrupt process', function() {
           isTTY: true,
           on: td.function(),
           setRawMode: td.function(),
+          removeListener: td.function(),
         },
       });
 
       let windowsCtrlCTrap = td.matchers.isA(Function);
 
       willInterruptProcess.addHandler(cb);
-      td.verify(process.stdin.setRawMode(true));
       td.verify(process.stdin.on('data', windowsCtrlCTrap));
     });
 
+    it('adds and reverts rawMode on Windows', function() {
+      Object.defineProperty(process, 'platform', {
+        value: 'win',
+      });
+
+      Object.defineProperty(process, 'stdin', {
+        value: {
+          isRaw: false,
+          isTTY: true,
+          on: td.function(),
+          setRawMode: td.function(),
+          removeListener: td.function(),
+        },
+      });
+
+      willInterruptProcess.addHandler(cb);
+      td.verify(process.stdin.setRawMode(true));
+
+      willInterruptProcess.removeHandler(cb);
+      td.verify(process.stdin.setRawMode(false));
+
+      td.verify(process.stdin.setRawMode(), {
+        ignoreExtraArgs: true,
```

Comment:
Ah yeah I see what you're doing.  I don't know how to make this any clearer with `testdouble` either.  This seems 👍  then.